src/sql-parser/src/parser.rs

```diff
@@ -1825,6 +1826,74 @@ impl<'a> Parser<'a> {
         };
         Ok(compression)
     }
+    /// Parse set of table definitions out of literal string from within CREATE SOURCE
+    /// 'tbl1_oid AS tbl1_name (col1_name col1_type, col2_name col2_type), tbl2_oid AS tbl2_name (col1_name col1_type,...)'
+    fn parse_source_tables(&mut self) -> Result<Vec<PgTable<Raw>>, ParserError> {
+        let mut tables: Vec<PgTable<Raw>> = Vec::new();
+        let raw_def = self.parse_literal_string()?;
+        if raw_def.is_empty() {
+            return Ok(tables);
+        }
+        let tokens = lexer::lex(&raw_def)?;
+        let mut parser = Parser::new(&raw_def, tokens);
+        loop {
+            let tbl_name = parser.parse_object_name()?;
+
+            let (col_def, _constaints) = match parser.parse_columns(Optional) {
+                Ok((col_def, _constraints)) => (col_def, _constraints),
+                Err(e) => {
+                    return Err(ParserError::new(
+                        parser.peek_pos(),
+                        format!(
+                            "Failed to parse columns for table {} due to: {}",
+                            tbl_name, e
+                        ),
+                    ))
+                }
+            };
+            parser.expect_keyword(WITH)?;
+            let opts: Vec<WithOption> = parser.parse_with_options(true)?;
+
+            let oid_val = opts.into_iter().find_map(|elem| {
+                if elem.key.to_string() == "oid" {
+                    match elem.value {
+                        Some(v) => Some(v),
+                        None => None,
+                    }
+                } else {
+                    None
+                }
+            });
+            let oid = match oid_val {
+                Some(val) => match u32::from_str(&val.to_string()) {
+                    Ok(i) => i,
+                    Err(e) => {
+                        return Err(ParserError::new(
+                            parser.peek_pos(),
+                            format!("Error parsing OID for table {}: {}", tbl_name, &e),
+                        ));
+                    }
+                },
+                None => {
+                    return Err(ParserError::new(
+                        parser.peek_pos(),
+                        format!("Missing WITH option oid for table: {}", tbl_name),
+                    ));
+                }
+            };
```

Comment:
If we're inventing this syntax, I'd just go with something simpler like `table (cols...) OID <oid>`. The parser's not meant to be peeking inside WITH option clauses (see long discussion in #eng-sql-coord recently).

Reply:
The primary reason I opted against this approach was based upon my understanding that `OID` would need to become a keyword which is currently is not. I followed along in that discussion as it happened and I actually came away with the understanding that in this case using with options was more SQL-y compared to adding a new keyword for this one specific case.
I had originally proposed `oid AS table_name (cols...)` since to my admittedly old school learning of SQL this was the most SQL-y because in practice we use the upstream OID in the `WHERE` clause when implementing materialized views from this materialized source.